src/external_trait_impls/rayon/raw.rs

```diff
@@ -0,0 +1,539 @@
+/// [`HashMap`]: /hashbrown/struct.HashMap.html
+pub struct ParValues<'a, K: 'a, V: 'a> {
+    table: &'a RawTable<(K, V)>,
+}
+
+impl<K, V: Sync> RawTable<(K, V)> {
+    #[inline]
+    pub fn par_values(&self) -> ParValues<K, V> {
+        ParValues { table: self }
+    }
+}
+
+unsafe impl<'a, K, V: Sync> Send for ParValues<'a, K, V> {}
+
+impl<'a, K, V: Sync> ParallelIterator for ParValues<'a, K, V> {
+    type Item = &'a V;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+    where
+        C: UnindexedConsumer<Self::Item>,
+    {
+        let buckets = unsafe { self.table.iter().iter };
+        let producer = ParValuesProducer::from(buckets);
+        plumbing::bridge_unindexed(producer, consumer)
+    }
+}
+
+impl<'a, K, V> Clone for ParValues<'a, K, V> {
+    fn clone(&self) -> Self {
+        ParValues { table: &*self.table }
+    }
+}
+
+impl<'a, K, V: fmt::Debug> fmt::Debug for ParValues<'a, K, V> {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let iter = unsafe { self.table.iter() };
+        let entries = iter.map(|bucket| unsafe {
+            &bucket.as_ref().1
+        });
+
+        f.debug_list().entries(entries).finish()
+    }
+}
+
+struct ParValuesProducer<'a, K: 'a, V: 'a> {
+    iter: RawIterRange<(K, V)>,
+    marker: PhantomData<&'a ()>,
+}
+
+impl<'a, K, V> From<RawIterRange<(K, V)>> for ParValuesProducer<'a, K, V> {
+    #[inline]
+    fn from(iter: RawIterRange<(K, V)>) -> Self {
+        Self { iter, marker: PhantomData }
+    }
+}
+
+unsafe impl<'a, K, V: Sync> Send for ParValuesProducer<'a, K, V> {}
+
+impl<'a, K, V: Sync> UnindexedProducer for ParValuesProducer<'a, K, V> {
+    type Item = &'a V;
+
+    fn split(self) -> (Self, Option<Self>) {
+        unsafe { raw_iter_range_split_and_convert(self.iter) }
+    }
+
+    fn fold_with<F>(self, folder: F) -> F
+    where
+        F: Folder<Self::Item>,
+    {
+        let iter = self.iter.map(|bucket| unsafe {
+            &bucket.as_ref().1
+        });
+        folder.consume_iter(iter)
+    }
+}
+
+
+/// Parallel iterator over mutable references to entries in a map.
+///
+/// This iterator is created by the [`par_iter_mut`] method on [`HashMap`]
+/// (provided by the [`IntoParallelRefMutIterator`] trait).
+/// See its documentation for more.
+///
+/// [`par_iter_mut`]: /hashbrown/struct.HashMap.html#method.par_iter_mut
+/// [`HashMap`]: /hashbrown/struct.HashMap.html
+/// [`IntoParallelRefMutIterator`]: https://docs.rs/rayon/1.0/rayon/iter/trait.IntoParallelRefMutIterator.html
+pub struct ParIterMut<'a, K: 'a, V: 'a> {
+    table: &'a mut RawTable<(K, V)>,
+}
+
+impl<'a, K: Sync, V: Send> IntoParallelIterator for &'a mut RawTable<(K, V)> {
+    type Item = (&'a K, &'a mut V);
+    type Iter = ParIterMut<'a, K, V>;
+
+    #[inline]
+    fn into_par_iter(self) -> Self::Iter {
+        ParIterMut { table: self }
+    }
+}
+
+unsafe impl<'a, K: Sync, V: Send> Send for ParIterMut<'a, K, V> {}
```

Comment:
This is not exactly implied by `&'a mut RawTable<(K, V)>`, but it is good enough. Remove this impl.